src/libsystemd-network/sd-ipv4ll.c

```diff
@@ -274,8 +274,14 @@ static int ipv4ll_start_internal(sd_ipv4ll *ll, bool reset_generation) {
         if (!ll->seed_set)
                 ll->seed.value = htole64(siphash24(ll->mac.ether_addr_octet, ETH_ALEN, MAC_HASH_KEY.bytes));
 
-        if (reset_generation)
+        if (reset_generation) {
                 ll->seed.generation = 0;
+                if (start_address) {
+                        r = sd_ipv4ll_set_address(ll, start_address);
+                        if (r < 0)
+                                return r;
```

Comment:
Funktioniert das richtig, wenn ich `IPV4LLStartAddress=` in die Konfiguration schreibe?
Dann ist `start_address != NULL`, aber `start_address->s_addr == 0` und `sd_ipv4ll_set_address() == -EINVAL`.

Der Fall `start_address == NULL` tritt über die Konfiguration ja nicht auf. Den gibt es nur künstlich in `sd_ipv4ll_restart`.

Reply:
Das muss ich heute noch mal ausprobieren, aber denke, dass es so besser wäre

`if (start_address && start_address->s_addr != 0) {`

Reply:
```
lan0: Acquiring IPv4 link-local address
Assertion 'ipv4ll_address_is_valid(address)' failed at src/libsystemd-network/sd-ipv4ll.c:230, function sd_ipv4ll_set_address(). Ignoring.
lan0: IPv4LL: Picked new IP address 169.254.55.202.
```